src/neuralnet/greylist.h

```diff
@@ -13,6 +13,8 @@ enum class GreylistReason
     ZERO_CREDIT_DAYS,  //!< Exported too many recent zero-credit days.
 };
 
+typedef std::vector<GreylistReason> GreylistReasons; //!< There can be more than one reason.
```

Comment:
Might be interesting to make a bitfield for `GreylistReason`s and define each reason as one bit position in that field. Then we can just check `== 0` for `GreylistReason::NONE`. Maybe for later... perhaps for now an empty `GreylistReasons` can represent `NONE` so we don't need to visit each reason to determine that.

Reply:
I thought about that too. Eliminating the GreylistReason::NONE and requiring that none is implictly defined by the absence of an element in the vector of reasons drastically simplifies the filter to basically checking whether reasons.size() >0.